src/journal/cat.c

```diff
@@ -123,14 +141,22 @@ static int run(int argc, char *argv[]) {
         if (r <= 0)
                 return r;
 
-        fd = sd_journal_stream_fd(arg_identifier, arg_priority, arg_level_prefix);
-        if (fd < 0)
-                return log_error_errno(fd, "Failed to create stream fd: %m");
+        outfd = sd_journal_stream_fd(arg_identifier, arg_priority, arg_level_prefix);
+        if (outfd < 0)
+                return log_error_errno(outfd, "Failed to create stream fd: %m");
+
+        if (arg_stderr_priority == arg_priority) {
+                errfd = outfd;
+        } else {
```

Comment:
coding style nitpick: we usually avoid the {} around single-line if blocks, also see CODING_STYLE